tests/virt_autotest/virt_autotest_base.pm

```diff
@@ -186,10 +201,16 @@ sub upload_virt_logs {
     my ($log_dir, $compressed_log_name) = @_;
 
     my $full_compressed_log_name = "/tmp/$compressed_log_name.tar";
+    if (check_var('ARCH', 's390x')) {
+        type_string "tar cvf $full_compressed_log_name $log_dir; gzip -f $full_compressed_log_name; rm $log_dir -r\n";
+        upload_asset"$full_compressed_log_name.gz", 1, 1;
+    }
+    else {
```

Comment:
As above, I recommend to not repeat all statements. This is not very maintainable. At least you could extract the actual command into a string and call either `type_string` or `script_run` but on the same command string

Reply:
Yes, this is good idea. 
Had made code changed as your mentioned, after local verification run, will update this PR,ASAP.

Reply:
Made code changed as your mentioned, and verification run as below:
S390 GUEST INSTALLATION SLES15SP1
http://10.67.17.251/tests/90
X86 GUEST INSTALLATION SLES15SP1
http://10.67.17.251/tests/17
X86 HOST UPGRADE SLES15->SLES15SP1
http://10.67.17.251/tests/32